### PR TITLE
chore: add jest config for ui package, convert Button tests to ts and fix existing assertions

### PR DESCRIPTION
**Pull request by brandongregoryscott**
## What kind of change does this PR introduce?

Chore - reinstates tests for the `ui` package. 

## What is the current behavior?

See https://github.com/supabase/supabase/pull/13540#issuecomment-1501920635 - it sounds like tests were started at some point (maybe in the https://github.com/supabase/ui repo) but weren't maintained once ported back into the monorepo. 

## What is the new behavior?

This PR adds a minimal jest config + refactors the `Button` component tests to pass based on its current code. It also adds a Github Action for running the tests in the `packages/ui` directory when any file in it changes. 

```
 PASS  src/components/Button/Button.test.tsx
  #Button
    ✓ should render button correctly (37 ms)
    ✓ should render different text (24 ms)
    ✓ should ignore events when disabled (16 ms)
    ✓ should ignore events when loading (20 ms)
    ✓ should have "w-full" class when block is true (68 ms)
    ✓ should have "primary" class from theme (29 ms)
    ✓ should have "default" class from theme (20 ms)
    ✓ should have "secondary" class from theme (9 ms)
    ✓ should have "alternative" class from theme (11 ms)
    ✓ should have "outline" class from theme (13 ms)
    ✓ should have "dashed" class from theme (14 ms)
    ✓ should have "link" class from theme (18 ms)
    ✓ should have "text" class from theme (13 ms)
    ✓ should have "danger" class from theme (5 ms)
    ✓ should have "warning" class from theme (6 ms)
    ✓ should have "tiny" class from theme (15 ms)
    ✓ should have "small" class from theme (8 ms)
    ✓ should have "medium" class from theme (7 ms)
    ✓ should have "large" class from theme (6 ms)
    ✓ should have "xlarge" class from theme (9 ms)
```

## Additional context

In an effort to keep this PR smaller and incrementally reinstate test suites / add new ones, I restricted the `testRegex` to just *.test.tsx files. I left a comment in the config file on why, but the existing tests are likely going to need some refactoring to pass anyway.

This PR also introduces a change to the way the `Button` component exposes its ref, since it was causing issues when wrapped with the Next `Link` component - I added a test for this as well. It's kinda hard to tell where/if the current ref was being used, but I think this should work.

### CODE REVIEW

File: packages/ui/src/components/Button/Button.test.tsx [hunk 1]
```diff
@@ -88,4 +89,24 @@ describe('#Button', () => {
 
     expect(screen.queryByRole('button')).toHaveClass(expected)
   })
+
+  it("shouldn't crash when wrapped with next/link", () => {
+    expect(() =>
+      render(
+        <Link href="https://supabase.com">
```
Comment (brandongregoryscott) [inline packages/ui/src/components/Button/Button.test.tsx:96]: This test was failing with before the change to the ref - doesn't seem related to passing a custom element with the `as` prop as originally theorized in https://github.com/supabase/supabase/pull/13540#issuecomment-1505152745

**Comment by saltcod** [comment]
Hey @brandongregoryscott ! had a look at this — your ui tests look good! But studio tests are failing on this branch. Any ideas? 

CleanShot 2023-05-15 at 17 01 40@2x

**Comment by brandongregoryscott** [comment]
Hey @saltcod - the studio tests are passing for me locally. Looks like they failed in Github Actions but not for the same reason [(JavaScript heap error)](https://github.com/supabase/supabase/actions/runs/4896703579/jobs/8922622104#step:5:365)

Those tests look time based - do we think they're just flaky? Not sure I have permission to retry the tests

**Comment by brandongregoryscott** [comment]
Following-up: This was happening on my fork off of the fresh `master`, too. Not sure why it's happening all of a sudden, could just be a tipping point. Following the instructions in this [comment](https://github.com/jestjs/jest/issues/9980#issuecomment-1124371891), this seems to resolve the CI issues: https://github.com/brandongregoryscott/supabase/actions/runs/4986366186/jobs/8927026785

**Review by saltcod — APPROVED** [review]

**Pull request merged**